src/robusta/integrations/prometheus/utils.py

```diff
@@ -26,195 +24,82 @@
-        response = prom._session.get(
-            f"{prom.url}/api/v1/query",
-            verify=prom.ssl_verification,
-            headers=prom.headers,
-            # This query should return empty results, but is correct
-            params={"query": "example", **params},
+    baseconfig = {
+        "url": url,
+        "disable_ssl": not PROMETHEUS_SSL_ENABLED,
+        "additional_labels": prometheus_params.prometheus_additional_labels,
+        "prometheus_url_query_string": prometheus_params.prometheus_url_query_string,
+    }
+    # aws config
+    if AWS_ACCESS_KEY:
+        return AWSPrometheusConfig(
+            access_key=AWS_ACCESS_KEY,
+            secret_access_key=AWS_SECRET_ACCESS_KEY,
+            service_name=AWS_SERVICE_NAME,
+            aws_region=AWS_REGION,
+            **baseconfig,
         )
-
-        if response.status_code == 401:
-            if PrometheusAuthorization.request_new_token():
-                prom.headers = PrometheusAuthorization.get_authorization_headers()
-                response = prom._session.get(
-                    f"{prom.url}/api/v1/query",
-                    verify=prom.ssl_verification,
-                    headers=prom.headers,
-                    params={"query": "example", **params},
-                )
-
-        response.raise_for_status()
-    except (ConnectionError, HTTPError) as e:
-        raise PrometheusNotFound(
-            f"Couldn't connect to Prometheus found under {prom.url}\nCaused by {e.__class__.__name__}: {e})"
-        ) from e
-
-
-def __text_config_to_dict(text: str) -> Dict:
-    conf = {}
-    lines = text.strip().split("\n")
-    for line in lines:
-        key, val = line.strip().split("=")
-        conf[key] = val.strip('"')
-
-    return conf
-
-
-def get_prometheus_flags(prom: "PrometheusConnect") -> Optional[Dict]:
-    try:
-        return fetch_prometheus_flags(prom=prom)
-    except Exception as e:
-        prometheus_exception = e
-
-    try:
-        return fetch_victoria_metrics_flags(vm=prom)
-    except Exception as e:
-        victoria_metrics_exception = e
-
-    raise PrometheusFlagsConnectionError(
-        f"Couldn't connect to the url: {prom.url}\n\t\tPrometheus: {prometheus_exception}"
-        f"\n\t\tVictoria Metrics: {victoria_metrics_exception})"
-    )
-
-
-def fetch_prometheus_flags(prom: "PrometheusConnect") -> Dict:
-    try:
-        result = {}
-
-        # connecting to prometheus
-        response = prom._session.get(
-            f"{prom.url}/api/v1/status/flags",
-            verify=prom.ssl_verification,
-            headers=prom.headers,
-            # This query should return empty results, but is correct
-            params={},
+    # coralogix config
+    if CORALOGIX_PROMETHEUS_TOKEN:
+        return CoralogixPrometheusConfig(**baseconfig, prometheus_token=CORALOGIX_PROMETHEUS_TOKEN)
+    # Azure config
+    if os.environ.get("AZURE_USE_MANAGED_ID"):
```

Comment:
The azure auth app flow is missing here, need to use something like that to support both flows.
`if os.environ.get("AZURE_USE_MANAGED_ID") or os.environ.get("AZURE_CLIENT_SECRET"):`